src/components/CytoscapeGraph/CytoscapeGraph.tsx

```diff
@@ -147,15 +148,11 @@ export class CytoscapeGraph extends React.Component<CytoscapeGraphProps, Cytosca
   };
 
   private turnEdgeLabelsTo = (cy: any, value: EdgeLabelMode) => {
-    cy.edges().forEach(edge => {
-      edge.data('edgeLabelMode', value);
-    });
+    cy.scratch(CytoscapeGlobalScratchNamespace).edgeLabelMode = value;
   };
 
   private turnNodeLabelsTo = (cy: any, value: boolean) => {
-    cy.nodes().forEach(node => {
-      node.data('showNodeLabels', value);
-    });
+    cy.scratch(CytoscapeGlobalScratchNamespace).showNodeLabels = value;
```

Comment:
I was going to say that we don't need this one anymore, but then i recall there was a bug, that the fix was to turn this on during a layout and set it to the desired value after all the layout magic is done.

Reply:
we still need this because GraphStyles now uses this to know if node labels should be on or off.